### ISSUE TITLE
Allow additional behaviour based on maximized state

### ISSUE THREAD

**Issue by Uldreth** [mlutfy/hidetopbar#158]
I would like to request a feature, if it is feasible and you are still working on this extension.

Basically, it would be really cool if it would be possible to set up this extension such that

1) the top panel hides only when obstructed by a _maximized_ window;

2) while in this mode, the top panel behaves in a "solid" manner in the sense that it will block non-maximized windows, and non-maximized windows will not try to take up space that is occupied by the panel.

In other words, the panel should behave exactly as it does by default (without this extension) _unless_ the _currently focused window_ is maximized, in which case, the autohide behaviour triggers. As far as I know, MacOS' top panel behaves in a similar way.

Thank you for giving it a thought.

**Comment by loaden** [comment]
I am looking for this behavior for a long time.
